refactor: sensor multilevel constants

chilicheech opened the pull request:
This refactors how we determine multilevel sensor types to be similar to how we do alarm sensors. It includes the names for the object ids.

@robertsLando there are some cases where the object id has the same name as the sensor type, so I kept it blank to avoid names like `direction_direction`. Let me know what you think.

Also, some sensors look like they belong to other types, but I didn't want to move them around just yet. For example, sensor 57 is under *sensor_water*, but it's a water pressure sensor, and maybe it should be under *sensor_pressure* instead..

Another thing to consider is in the future we may want to populate not just object ids but also units of measurement based on the constants.

It would be really great if instead of redefining things in the Constants.js if we could just read in the open-zwave xml files that it's based on and parse them to the correct attributes, i.e.: config/Localization.xml and config/SensorMultiLevelCCTypes.xml

Cheers,

robertsLando commented on lib/Constants.js:
```diff
@@ -32,92 +32,125 @@ module.exports = {
+
+    for (let sensor in sensorMap) {
+      let objectId = sensorMap[sensor][index]
+      if (objectId === null) continue // objectId can be an empty string
```
This should be `objectId === undefined`

robertsLando commented on lib/Constants.js:
```diff
@@ -32,92 +32,125 @@ module.exports = {
-        return 'sensor_heart'
-      default:
-        return null
+    let sensorMap = {
```
There are some more sensors in the list, also I suggest to add another map with sensors icons so we can keep only one sensor generic configuration and edit the object_id and the discovery payload mdi icon

chilicheech commented:
I added the missing sensors and changed null to undefined.
As for the other map for the icons, I think it would be better to do that in a separate PR. We could also do something similar for the units of measurement.

The more I look at this the more I think we should just read in the config/NotificationCCTypes.xml and config/SensorMultiLevelCCTypes.xml files and parse them into objects we can use. They have `id` and `name` that we can use. Name can be changed to lower case and special characters changed to `_`. And for the sensor multilevel we could even use the first unit of measurement in the list.

This way we don't have to keep our own maps, and just use what's supplied by OZW. The only map we would need is for the icons.

robertsLando commented:
@chilicheech I understand what you mean but I prefer to keep my own map for now, the best thing would be getting them from OZW directly instead of reading xml files but that requires a change in openzwave-shared library

robertsLando commented:
> As for the other map for the icons, I think it would be better to do that in a separate PR. We could also do something similar for the units of measurement.

I prefer to merge this once we clean up all sensors, there is still lot confusion in configuration.js file too many repeating configurations. About units of measurements them are overrided by valueId units so they can all be removed from configurations

Thanks for your help @chilicheech

chilicheech commented:
Thanks. I saw the modifications you made with the `props` field. Looks good.

Cheers,

robertsLando commented:
Thanks to you! :)

Daniel

> On 14 Feb 2020, at 18:47, Thiago Oliveira <notifications@github.com> wrote:
> 
> ﻿
> Thanks. I saw the modifications you made with the props field. Looks good.
> 
> Cheers,
> 
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub, or unsubscribe.
